packages/gatsby-source-drupal/src/gatsby-node.ts

```diff
@@ -951,10 +952,22 @@ exports.createSchemaCustomization = (
   if (pluginOptions.skipFileDownloads && pluginOptions.imageCDN) {
     actions.createTypes([
       // polyfill so image CDN works on older versions of Gatsby
+      // this type is merged in with the inferred file__file and files types, adding Image CDN support via the gatsbyImage GraphQL field. The `RemoteFile` interface as well as the polyfill above are what add the gatsbyImage field.
       addRemoteFilePolyfillInterface(
-        // this type is merged in with the inferred file__file type, adding Image CDN support via the gatsbyImage GraphQL field. The `RemoteFile` interface as well as the polyfill above are what add the gatsbyImage field.
         schema.buildObjectType({
-          name: `file__file`,
+          name: generateTypeName(`file__file`, pluginOptions.typePrefix),
```

Comment:
Would using `generateTypeName` change the name of the type?

It uses this:

```
prefixed.replace(/-|__|:|\.|\s/g, `_`)
```

So wouldn't this be converted to `file_file`?

Reply:
Oooh, good point. We use `file--file` elsewhere for that reason. I would really like `generateTypeName` to be idempotent but unfortunately it would be a breaking change